### PR TITLE
CMP: Fix TLS hostname checking and out-of-memory error reporting

### PR DESCRIPTION
**Pull request by DDvO**
* `apps/cmp.c`: Fix TLS hostname checking in case `-server` provides URL (i.e, more than just a hostname)
* `OSSL_CMP_CTX_new()`: Fix distinction of out-of-memory and other errors

### CODE REVIEW

File: apps/cmp.c [hunk 1]
```diff
@@ -1311,8 +1312,7 @@ static SSL_CTX *setup_ssl_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
     if (opt_tls_trusted != NULL) {
         /* enable and parameterize server hostname/IP address check */
         if (!truststore_set_host_etc(trust_store,
-                                     opt_tls_host != NULL ?
-                                     opt_tls_host : opt_server))
+                                     opt_tls_host != NULL ? opt_tls_host : host))
             /* TODO: is the server host name correct for TLS via proxy? */
```
Comment (paulidale) [inline apps/cmp.c:1316]: Have you have resolved this TODO comment?  If so, please remove it too.

**Review by paulidale — APPROVED** [review]
Just a query about a comment, good with or without it.

File: apps/cmp.c (hunk 1 above)
Comment (DDvO) [reply apps/cmp.c:1316]: Thanks Pauli for your swift approval!
I also noticed this somewhat related comment. Unfortunately the answer regarding the proxy host name is still open to me.

**Comment by DDvO** [comment]
Merged - thanks

**Pull request closed without merging**